File: CS/E3252_CS/Controllers/ServerModeEFController.cs
using System;
using System.Web.Mvc;

/*DevExpress.Xpf.Core.vXX.Y.dll*/
using DevExpress.Data.Linq;
/*DevExpress.Xpf.Core.vXX.Y.dll*/

namespace E3252_CS.Controllers {
    public class ServerModeEFController : Controller {
        public ActionResult IndexEF() {
            return View(GetEntityServerModeSource());
        }
        public ActionResult GridViewPartialEF() {
            return PartialView(GetEntityServerModeSource());
        }
        private EntityServerModeSource GetEntityServerModeSource() {
            EntityServerModeSource esms = new EntityServerModeSource();
            esms.KeyExpression = "ProductID";
            esms.QueryableSource = new NorthwindEntities().Products;

            return esms;
        }
    }
}

File: VB/E3252_VB/Controllers/ServerModeEFController.vb
Imports System
Imports System.Web.Mvc

'DevExpress.Xpf.Core.vXX.Y.dll
Imports DevExpress.Data.Linq
'DevExpress.Xpf.Core.vXX.Y.dll

Public Class ServerModeEFController
    Inherits Controller
    Public Function IndexEF() As ActionResult
        Return View(GetEntityServerModeSource())
    End Function
    Public Function GridViewPartialEF() As ActionResult
        Return PartialView(GetEntityServerModeSource())
    End Function
    Private Function GetEntityServerModeSource() As EntityServerModeSource
        Dim esms As New EntityServerModeSource()
        esms.KeyExpression = "ProductID"
        esms.QueryableSource = New NorthwindEntities().Products

        Return esms
    End Function
End Class
